backend/de.metas.handlingunits.base/src/main/java/de/metas/manufacturing/issue/plan/PPOrderIssuePlanCreateCommand.java

```diff
@@ -157,28 +228,39 @@ private Stream<PPOrderIssuePlanStep> createSteps(final I_PP_Order_BOMLine orderB
 					allocableHU.addQtyAllocated(huQtyAvailable);
 					allocatedQty = allocatedQty.add(huQtyAvailable);
 				}
-
 				planSteps.add(planStep);
 			}
-		}
 
-		final Quantity qtyToAllocate = targetQty.subtract(allocatedQty);
-		if (qtyToAllocate.signum() > 0)
-		{
-			throw new AdempiereException(MSG_CannotFullAllocate,
-					productBL.getProductName(productId),
-					warehouseBL.getLocatorNameById(pickFromLocatorId),
-					qtyToAllocate,
-					targetQty)
-					.markAsUserValidationError();
+			if (createStepsForIncludedTUs)
+			{
+				streamIncludedTUs(allocableHU)
+						.map(tu -> PPOrderIssuePlanStep.builder()
+								.orderBOMLineId(orderBOMLineId)
+								.productId(productId)
+								.qtyToIssue(targetQty.toZero())
+								.pickFromLocatorId(allocableHU.getLocatorId())
+								.pickFromTopLevelHU(tu)
+								.isAlternative(true)
+								.build())
+						.forEach(planSteps::add);
+			}
 		}
 
-		return planSteps.stream();
+		return allocatedQty;
+	}
+
+	private boolean shouldConsiderHUsFromTheWholeWarehouse()
```

Comment:
should again :)
personally, when I read "should" I understand it as a recommendation, something that is not a must but it's better,
while "is" is black/white to me.